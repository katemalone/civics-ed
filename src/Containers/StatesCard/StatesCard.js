import React, { Component } from 'react';
import AK from '../../utils/Images/AK.svg';
import AL from '../../utils/Images/AL.svg';
import AR from '../../utils/Images/AR.svg';
import AZ from '../../utils/Images/AL.svg';
import CA from '../../utils/Images/CA.svg';
import CO from '../../utils/Images/CO.svg';
import { getStateInfo } from '../../utils/apiCalls';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { addStateInfo } from '../../Actions'
import { Redirect } from 'react-router-dom';


export class StatesCard extends Component {
  constructor(){
    super();
    this.state = {
      isClicked : false 
    }
  }

  
  handleClick = (e) => {
    e.preventDefault()
    // let state = e.target.parentNode.id;
    // getStateInfo(state)
    this.setState({isClicked: true })

  }

  render(){
  const images = { AK, AL, AR, AZ, CA, CO };
  const { id, name, statesImg } = this.props;
  
    if(this.state.isClicked){ 
    return <Redirect to='/stateInfo' />
    } 
  return ( 
    <section className="StatesCard" id={id}>
      <div className="StatesCard-div">
      <div className="StatesCard_header">
      <h2 className="header_name" >{name}</h2>
      <p className="header_description">Choose {name} to see the representatives , bills, and something else</p>
      </div>
      <div className="StatesCard_img-div">
      <i className="fa fa-arrow-left arrow" />
      <img className="img_img" src={images[statesImg]} /> 
      <i className="fa fa-arrow-right arrow" />
      </div>
      </div>
      <button className="StatesCard_button" onClick={(e)=> this.handleClick(e)}>Learn More!</button>
    </section>
  )
  }
}


// export const mapStateToProps = ({}) => ({
  
// })

export const mapDispatchToProps = dispatch => {
  bindActionCreators({
    addStateInfo
  }, dispatch)
}

export default connect(null, mapDispatchToProps)(StatesCard)
