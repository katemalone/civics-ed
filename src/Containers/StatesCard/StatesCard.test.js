import React from 'react';
import { shallow } from 'enzyme';
import { StatesCard } from './StatesCard';

describe('StatesCard', () => {


  it('should match snapShot', () => {
    const wrapper = shallow(
      <StatesCard />);
    expect(wrapper).toMatchSnapshot()
  });
});