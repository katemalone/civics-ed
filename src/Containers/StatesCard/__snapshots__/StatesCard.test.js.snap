// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatesCard should match snapShot 1`] = `
<section
  className="StatesCard"
>
  <div
    className="StatesCard-div"
  >
    <div
      className="StatesCard_header"
    >
      <h2
        className="header_name"
      />
      <p
        className="header_description"
      >
        Choose 
         to see the state representatives. 
      </p>
    </div>
    <div
      className="StatesCard_img-div"
    >
      <i
        className="fa fa-arrow-left arrow"
      />
      <img
        className="img_img"
      />
      <i
        className="fa fa-arrow-right arrow"
      />
    </div>
    <button
      className="StatesCard_button"
      onClick={[Function]}
    >
       Choose 
      ! 
    </button>
  </div>
</section>
`;
